File: src/test/java/com/ddd/trainings/infra/InMemoryTrainingProposalRepo.java
package com.ddd.trainings.infra;

import com.ddd.trainings.domain.ProposalId;
import com.ddd.trainings.domain.TrainingProposal;
import com.ddd.utils.InMemoryRepo;

public class InMemoryTrainingProposalRepo
    extends InMemoryRepo<TrainingProposal, ProposalId>
    implements TrainingProposalRepo {

}

File: src/Ddd.Tests/Trainings/Proposal/Adapters/InMemoryTrainingProposalRepo.cs
using System.Collections.Generic;
using System.Linq;
using Ddd.Trainings.Proposal.Domain;
using Ddd.Trainings.Proposal.Domain.Ports;
using Ddd.Utils;

namespace Ddd.Tests.Trainings.Proposal.Infra.Tests;

public class InMemoryTrainingProposalRepo : BaseRepo<TrainingProposal, ProposalId>, ITrainingProposalRepo
{
    public override List<TrainingProposal> FindAll()
    {
        return base.FindAll().ToList();
    }
}